client/my-sites/patterns/components/pattern-preview/index.tsx

```diff
@@ -84,6 +87,48 @@ function PatternPreviewFragment( {
 		ref.current?.dispatchEvent( new CustomEvent( 'patternPreviewResize', { bubbles: true } ) );
 	}, [ nodeSize.width, nodeSize.height ] );
 
+	// We deliberately use `window.scrollBy` instead of setting an ID attribute on the
+	// `.pattern-preview` element to avoid a janky experience for users while the page is loading.
+	// This way, the browser doesn't scroll down to the relevant patterns until patterns are mostly
+	// finished loading.
+	useEffect( () => {
+		if (
+			window.location.hash !== `#${ idAttr }` ||
+			hasScrolledToAnchorRef.current ||
+			! ref.current
+		) {
+			return;
+		}
+
+		const element = ref.current;
+
+		const timeoutId = setTimeout( function () {
+			hasScrolledToAnchorRef.current = true;
+
+			const masterbarHeightRaw = getComputedStyle( document.documentElement ).getPropertyValue(
+				'--masterbar-height'
+			);
+			const masterbarHeight = /^\d+px$/.test( masterbarHeightRaw )
+				? parseInt( masterbarHeightRaw )
+				: 0;
+
+			const stickyNav = document.querySelector( '.pattern-library__pill-navigation' );
+			const stickyNavCoords = stickyNav?.getBoundingClientRect();
+			const stickyNavHeight = stickyNavCoords && ! isMobile() ? stickyNavCoords.height : 0;
+
+			const elementCoords = element.getBoundingClientRect();
+
+			window.scrollBy( {
+				behavior: 'smooth',
+				top: elementCoords.top - stickyNavHeight - masterbarHeight - 16,
+			} );
```

Comment:
I'd like your take on this, @nightnei, compared to `Element.scrollIntoView`. This still doesn't yield perfect results in Safari and Firefox (likely because the iframes are using `loading=lazy`, which means that the layout shifts while the browser scrolls), but it looks better in all browsers and essentially perfect in Chrome AFAICT.

The `querySelector` call isn't very React-y, but I think this is fine for this particular purpose. Passing `navRef` all the way down here helps mostly on paper.

Reply:
I am ok with intentional using of querySelector, and it works quite good now in all browsers - FF and Safari has bigger gap at the top, but it looks totally fine 👍